presentation-compiler/src/main/dotty/tools/pc/printer/ShortenedTypePrinter.scala

```diff
@@ -166,6 +165,37 @@ class ShortenedTypePrinter(
           res.toPrefixText
       }
 
+  /** Find a shortened prefix for a symbol which can be accessed statically in current run.
+   *  @param sym - statically accessible symbol
+   *
+   *  @return shortened prefix for sym
+   */
+  private def missingSymbolPrefixOf(sym: Symbol): Text =
+
+    def toTextPrefix0(owner: Symbol, acc: List[Symbol], wasPreviousAConflict: Boolean): List[Symbol] =
+      if !owner.exists then acc
+      else if owner.is(Synthetic) then toTextPrefix0(owner.maybeOwner, acc, wasPreviousAConflict)
+      else
+        indexedCtx.lookupSym(owner) match
+          case Result.Conflict => toTextPrefix0(owner.maybeOwner, owner :: acc, wasPreviousAConflict = true)
+          // Previous owner was at conflict so we have to include current one in prefix to disambiguate
+          case _ if wasPreviousAConflict => toTextPrefix0(owner.maybeOwner, owner :: acc, wasPreviousAConflict = false)
+          // If symbol is defined in the current run, we want to strip the symbol until first InScope is found
+          case Result.InScope if sym.isDefinedInCurrentRun => owner :: acc
```

Comment:
Do we want for things to behave differently if `isDefinedInCurrentRun` ? Also note that this won't behave like this if you have more than a single inner package, because you'll fall into one of the cases lower.